src/compiler/scala/tools/nsc/typechecker/PatternTypers.scala

```diff
@@ -97,9 +97,9 @@ trait PatternTypers {
       // Dueling test cases: pos/overloaded-unapply.scala, run/case-class-23.scala, pos/t5022.scala
       // A case class with 23+ params has no unapply method.
       // A case class constructor may be overloaded with unapply methods in the companion.
-      // A case class maybe have its own custom unapply (so non-synthetic) scala/bug#11252
-      // Unapply methods aren't `isCaseApplyOrUnapply` in Scala 3 tasty/run/src-2/tastytest/TestColour.scala
-      else if (canElide && caseClass.isCase && !member.isOverloaded && (member == NoSymbol || member.isSynthetic))
+      // A case class may have its own custom unapply (so non-synthetic) pos/t11252.scala
+      // A case class may have its own custom unapply unrelated to deconstructing pos/t12250.scala
+      else if (canElide && caseClass.isCase && (!member.isOverloaded && (member == NoSymbol || member.isSynthetic) || member.alternatives.exists(_.isSynthetic) && caseClass.tpe =:= pt))
```

Comment:
I don't know my way around here that well; IIUC this assumes that the synthetic `unapply` overload is the one selected - could it be that overloading resolution picks a different one, would it be fine to go in`convertToCaseConstructor`?

Reply:
I think there is only ever one synthetic unapply for a case class (`caseClass.isCase`).  And your understanding is correct, except it's not really "selected": the presence of the case class unapply means that using its constructor is the right course of action.

Now that I think about it, I should review the case-class-23 situation.  Basically it never had an unapply and we rewrote it to use its constructor.  What happens in the various situations of redefining its unapply or defining an un-related unapply? todo += 1

Reply:
OK, the situation is there's an issue (IMO) but thankfully my change here didn't affected it.